### PR TITLE
Fixes #22525: Directives applied twice don't show in rule details (they should be skipped)

### PR DESCRIPTION
**Pull request by ElaadF**
# :warning: Closed in favor of https://github.com/Normation/rudder/pull/5600

---
### :warning: Do not merge this PR, this is still in Work in Progress :warning: 

https://issues.rudder.io/issues/22525

### CODE REVIEW

File: webapp/sources/rudder/rudder-web/src/main/elm/rules/sources/ViewTabContent.elm [hunk 1]
```diff
@@ -294,20 +295,21 @@ directivesTab model details =
...
        Desc -> (List.reverse childrenSort, "desc", Asc)
 
     ruleDirectives = directives
       |> List.filter (\d -> List.member d.id ruleDirectivesId)
 
     disabledRuleDirectives = ruleDirectives
       |> List.filter (\d -> not d.enabled)
-
+    skippedDirectives = ruleDirectives
+      |> List.filter (\d -> (not (List.member d.id.value (List.map (\f -> f.directiveId.value) directivesChildren))) && d.enabled)
```
Comment (ElaadF) [inline webapp/sources/rudder/rudder-web/src/main/elm/rules/sources/ViewTabContent.elm:312]: Here I retrieve the list of directives that are enabled and that are absent from the compliance list of directives. In this bug, the problem is that the directive skipped is not part of the response of the API `getRulesComplianceId`.
But we have the skipped directive when we get the directive tree library.

So we can guess that if the directive is skipped:
- Enable
- present in directive tree
- linked to the `ruleId`
- absent from the directives compliance list of `ruleId`

File: webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/ComplianceApi.scala [hunk 2]
```diff
@@ -334,20 +336,27 @@ class ComplianceAPIService(
...
+      val overridesRules = overridesByRule.view.mapValues(_ => AggregatedStatusReport(Nil)).toMap
 
+      println(s"overridesByRule.toList ${overridesByRule.toList}")
+      println(s"overridesRules ${overridesRules.toList.foreach(println)}")
       val reportsByRule = reportsByNode.flatMap { case (_, status) => status.reports }.groupBy(_.ruleId)
+      val reportsByRule2 = reportsByNode.flatMap { case (_, status) => status.byRules }
       val t7            = System.currentTimeMillis()
       TimingDebugLoggerPure.logEffect.trace(s"getByRulesCompliance - group reports by rules in ${t7 - t6} ms")
-
+      val finalReports = overridesRules ++ reportsByRule2
```
Comment (ElaadF) [inline webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/ComplianceApi.scala:349]: I was trying to send the overridden (skipped) directives to send it to the `getRulesComplianceId` API, but it seems like is not working, we don't get the overridden directives in the response

**Comment by clarktsiory** [comment]
PR rebased

**Comment by clarktsiory** [comment]
PR rebased

**Comment by clarktsiory** [comment]
Closed in favor of https://github.com/Normation/rudder/pull/5600

**Pull request closed without merging**